internal/storage/bucket_handle_test.go

```diff
@@ -256,3 +257,102 @@ func (t *BucketHandleTest) TestCreateObjectMethodWhenGivenGenerationObjectNotExi
 	AssertEq(nil, obj)
 	AssertTrue(strings.Contains(err.Error(), "Error 412: Precondition failed"))
 }
+
+func (t *BucketHandleTest) TestGetProjectValueWhenGcloudProjectionIsNoAcl() {
+	proj := getProjectionValue(gcs.NoAcl)
+
+	AssertEq(storage.ProjectionNoACL, proj)
+}
+
+func (t *BucketHandleTest) TestGetProjectValueWhenGcloudProjectionIsFull() {
+	proj := getProjectionValue(gcs.Full)
+
+	AssertEq(storage.ProjectionFull, proj)
+}
+
+func (t *BucketHandleTest) TestGetProjectValueWhenGcloudProjectionIsDefault() {
+	proj := getProjectionValue(0)
+
+	AssertEq(storage.ProjectionFull, proj)
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithPrefixObjectExist() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "gcsfuse/",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertEq(nil, err)
+	AssertEq(TestObjectRootFolderName, obj.Objects[0].Name)
+	AssertEq(TestObjectSubRootFolderName, obj.Objects[1].Name)
+	AssertEq(TestObjectName, obj.Objects[2].Name)
+	AssertEq(TestObjectGeneration, obj.Objects[0].Generation)
+	AssertEq(TestObjectSubRootFolderName, obj.CollapsedRuns[0])
+	AssertEq(3, len(obj.Objects))
+	AssertEq(1, len(obj.CollapsedRuns))
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithPrefixObjectDoesNotExist() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "PrefixObjectDoesNotExist",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertEq(nil, err)
+	AssertEq(nil, obj.Objects)
+	AssertEq(nil, obj.CollapsedRuns)
+	AssertEq(0, len(obj.Objects))
```

Comment:
if obj.objects and obj.collapsedruns is nil, why to check for length.

Reply:
fixed